src/query/storages/fuse/src/io/read/block/parquet/mod.rs

```diff
@@ -0,0 +1,112 @@
+impl BlockReader {
+    pub(crate) fn column_chunks_to_data_block_1(
+        &self,
+        num_rows: usize,
+        column_metas: &HashMap<ColumnId, ColumnMeta>,
+        column_chunks: HashMap<ColumnId, DataItem>,
+        compression: &Compression,
+        block_path: &str,
+    ) -> databend_common_exception::Result<DataBlock> {
+        if column_chunks.is_empty() {
+            return self.build_default_values_block(num_rows);
+        }
+        let record_batch = deserialize_column_chunks(
+            &self.original_schema,
+            num_rows,
+            &column_chunks,
+            compression,
+        )?;
+        let mut columns = Vec::with_capacity(self.projected_schema.fields.len());
+        for (i, field) in self.projected_schema.fields.iter().enumerate() {
+            let data_type = field.data_type().into();
+            let leaf_column_ids = field.leaf_column_ids();
+            if leaf_column_ids.len() > 1
+                && leaf_column_ids
+                    .iter()
+                    .any(|id| matches!(column_chunks.get(id), Some(DataItem::ColumnArray(_))))
+            {
+                return Err(ErrorCode::StorageOther(
+                    "unexpected nested field: nested leaf field hits cached",
+                ));
+            }
+
+            let value = match column_chunks.get(&field.column_id) {
+                Some(DataItem::RawData(data)) => {
+                    let arrow_array = column_by_name(&record_batch, &field.name);
+                    let arrow2_array: Box<dyn databend_common_arrow::arrow::array::Array> =
+                        arrow_array.into();
+                    if self.put_cache && leaf_column_ids.len() == 1 {
+                        if let Some(cache) = CacheManager::instance().get_table_data_array_cache() {
+                            let meta = column_metas.get(&field.column_id).unwrap();
+                            let (offset, len) = meta.offset_length();
+                            let key =
+                                TableDataCacheKey::new(block_path, field.column_id, offset, len);
+                            cache.put(key.into(), Arc::new((arrow2_array.clone(), data.len())))
+                        }
+                    }
+                    Value::Column(Column::from_arrow(arrow2_array.as_ref(), &data_type)?)
+                }
+                Some(DataItem::ColumnArray(cached)) => {
+                    Value::Column(Column::from_arrow(cached.0.as_ref(), &data_type)?)
+                }
+                None => Value::Scalar(self.default_vals[i].clone()),
+            };
+            columns.push(BlockEntry::new(data_type, value));
+        }
+        Ok(DataBlock::new(columns, num_rows))
+    }
+}
+
+fn column_by_name(record_batch: &RecordBatch, name: &str) -> ArrayRef {
+    let arrow_array = record_batch
+        .column_by_name(name)
+        .cloned()
+        .unwrap_or_else(|| {
+            let names = name.split(':').collect::<Vec<_>>();
```

Comment:
Splitting names directly with `:` may be unsafe, we need a better way to handle nested struct name.

Reply:
Any detailed advice?

Reply:
Changes made, PTAL @b41sh